routing/fseg.hpp

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "coding/reader.hpp"
+#include "coding/write_to_sink.hpp"
+
+#include "std/cstdint.hpp"
+#include "std/limits.hpp"
+
+namespace routing
+{
+class FSegId final
+{
+  static uint32_t constexpr kInvalidFeatureId = numeric_limits<uint32_t>::max();
+
+public:
+  static FSegId MakeInvalid() { return FSegId(kInvalidFeatureId, 0); }
+
+  FSegId() = default;
```

Comment:
|m_featureId| and |m_segId| should be defined here. 
FSegId() : m_featureId(0), m_segId(0) {}
or
FSegId() : m_featureId(kInvalidFeatureId), m_segId(0) {}